homework/index.js

```diff
@@ -30,13 +32,126 @@
+            createAndAppend('td', tableRow, {
+              text: `${key} :`,
+              class: 'title',
+            });
+            createAndAppend('td', tableRow, { text: value });
+          }
+          if (key === 'updated') {
+            const tableRow = createAndAppend('tr', parentTable, { class: 'tr' });
+            createAndAppend('td', tableRow, {
+              text: 'updated :',
+              class: 'title',
+            });
+            createAndAppend('td', tableRow, {
+              text: `${value.slice(0, 10)}, ${value.slice(11, 19)}`,
+            });
+          }
+        }
+      });
+    } else {
+      parentTable.innerHTML = '';
+      showRepoDetailsInTable(obj, parentTable);
+    }
+  }
+
+  function showContributorsAsList(obj, parent) {
+    fetchJSON(obj.contributorsUrl, (error, data) => {
+      if (error) {
+        createAndAppend('div', parent, { text: error.message, class: 'alert-error' });
+      } else {
+        parent.innerHTML = '';
+        if (data === null) {
+          createAndAppend('div', parent, { text: 'No Contribution Data.', class: 'alert-error' });
+        } else {
+          createAndAppend('p', parent, { text: 'Contributions', class: 'contributors-header' });
+          const ul = createAndAppend('ul', parent, { class: 'contributors-list' });
+
+          data.forEach(contributor => {
+            const li = createAndAppend('li', ul, { class: 'contributor-item' });
+            createAndAppend('img', li, {
+              src: contributor.avatar_url,
+              alt: contributor.login,
+              class: 'contributor-avatar',
+            });
+            const contributorData = createAndAppend('div', li, { class: 'contributor-data' });
+            createAndAppend('div', contributorData, { text: contributor.login });
+            createAndAppend('div', contributorData, {
+              text: contributor.contributions,
+              class: 'contribution-count',
+            });
+            li.addEventListener('click', () => {
+              window.open(contributor.html_url);
+            });
+          });
+        }
+      }
+    });
+  }
+
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const header = createAndAppend('header', root, { class: 'header' });
+        createAndAppend('label', header, { for: 'select-menu', text: 'HYF Repositories' });
+        const selectMenu = createAndAppend('select', header, {
+          class: 'select-menu',
+          id: 'select-menu',
+        });
+
+        const sortedData = data.sort((a, b) =>
+          a.name.toLowerCase().localeCompare(b.name.toLowerCase()),
+        );
+
+        const elementsToUseInPage = sortedData.map(element => ({
```

Comment:
Since you haven't reserved to use of the name `repos` or `repositories` elsewhere, this seems a good place to use instead of  `elementsToUseInPage`. You could also chain the `sort` and `map` operations so that you can get rid of the `sortedData` intermediate variable:

```js
const repos = data
  .sort((a, b) => a.name.toLowerCase().localeCompare(b.name.toLowerCase()))
  .map(element => ({
    name: element.name,
    description: element.description,
    forks: element.forks,
    updated: element.updated_at,
    repoUrl: element.html_url,
    contributorsUrl: element.contributors_url,
  }));
```

Later on (line 139), your `forEach` can then become:

```js
repos.forEach((repo, index) => {
  ...
});
```